backend/lib/resolve-payload-cids.js

```diff
@@ -106,3 +117,31 @@ async function updatePayloadCidInActiveDeal (pgPool, deal, newPayloadRetrievalSt
     throw Error(util.format('Error updating payload of deal: ', deal), { cause: error })
   }
 }
+
+function getSmartContractClient () {
+  const fetchRequest = new ethers.FetchRequest(RPC_URL)
+  assert(GLIF_TOKEN, 'GLIF_TOKEN is required')
+  fetchRequest.setHeader('Authorization', `Bearer ${GLIF_TOKEN}`)
+  const provider = new ethers.JsonRpcProvider(fetchRequest)
+  return new ethers.Contract(
+    MINER_TO_PEERID_CONTRACT_ADDRESS,
+    MINER_TO_PEERID_CONTRACT_ABI,
+    provider
+  )
+}
+const defaultSmartContractClient = getSmartContractClient()
+
+/**
+ * @param {number} minerId
+ * @param {object} options
```

Comment:
```suggestion
 * @param {object} [options]
```

This is an optional parameter.

To me, wrong types are worse than no types (it adds confusion and misdirects you), therefore I find it important to have these be correct